src/main/java/com/mycompany/onprem/NtlmExtension.java

```diff
@@ -0,0 +1,107 @@
+package com.mycompany.onprem;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.NTCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+public class NtlmExtension {
+
+	private static final String HEADER_WORKATO_URL = "X-Workato-NTLM-url";
+
+	// username:password@domain/workstation
+	private static final String HEADER_WORKATO_AUTH = "X-Workato-NTLM-auth";
+
+	@Inject
+	private Environment env;
+
+	private CloseableHttpClient httpClient = HttpClients.createDefault();
+
+	@RequestMapping(path = "/request", method = RequestMethod.POST)
+	public @ResponseBody String request(@RequestHeader(HEADER_WORKATO_AUTH) String authHeader,
+										@RequestHeader(HEADER_WORKATO_URL) String urlHeader,
+										HttpServletResponse httpServletResponse) {
+
+		NtlmAuth auth = NtlmAuth.build(authHeader);
+
+		if (auth.username != null && auth.password != null) {
```

Comment:
@spn What is our tab space policy for JAVA code? (can it be 2 spaces?, 8 spaces makes reading the code in GH hard..

Reply:
@kandadaboggu we should use 4 spaces as per Sun coding standard